### PR TITLE
Reproduce runs

### PR DESCRIPTION
**Pull request by Blaizzy**
# Description

Include a summary of the changes and the related issue.

__Related to:__ `(ClickUp/JIRA task name)`

__Any expected test failures?__

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [x] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Precommit installed and run before pushing changes
- [x] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [ ] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS: MacOS Ventura 13.2.1
- Python version: 3.9
- Neptune version: 1.0.2
- Affected libraries with version:

### CODE REVIEW

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/reproduce_run.py:1]:
```suggestion
import os
import neptune
```

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/reproduce_run.py:6]:
```suggestion
os.environ["NEPTUNE_PROJECT"] = "common/showroom"
os.environ["NEPTUNE_API_TOKEN"] = "ANONYMOUS"
```

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/reproduce_run.py:22]:
```suggestion
# (Neptune) Fetch only inactive runs with tags "showcase-run", "reproduce", and "Basic script" in the project
with neptune.init_project(
    project=project_name, api_token=neptune.ANONYMOUS_API_TOKEN, mode="read-only"
) as project:
  runs_table_df = project.fetch_runs_table(
      state="inactive", tag=["showcase-run", "reproduce", "Basic script"]
  ).to_pandas()
```

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/reproduce_run.py:27]:
```suggestion
print(f"{old_run_id=}")
```

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/reproduce_run.py:40]:
```suggestion
old_run = neptune.init_run(
    with_id=old_run_id,
    mode="read-only",
)
```

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/reproduce_run.py:57]:
```suggestion
new_run = neptune.init_run(tags=["reproduce", "new-run"])
```

File: how-to-guides/reproduce-run/scripts/run_examples.sh
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/run_examples.sh:7]:
```suggestion
python reproduce_run.py

echo "Checkout to commit ID 146841173bb1f05b00c587f689695d041fdc55c3 ..." 
# We can get this commit ID from the source_code/git namespace of the run
git checkout 146841173bb1f05b00c587f689695d041fdc55c3

echo "Running old_run.py..."
python old_run.py
```

Merging both the scripts as these will mostly be used only for testing anyway

File: how-to-guides/reproduce-run/scripts/run_git_checkout.sh
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/scripts/run_git_checkout.sh:1]: We can remove this script after merging it with `run_examples.sh`

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
Some suggestions.
Good work overall 🎉

File: .github/workflows/test-scripts.yml
Comment (Blaizzy) [reply .github/workflows/test-scripts.yml:26]: You read my mind. I was thinking about it. 

But won't that be confusing and hard to follow? The shell scripts are for our testing. Users normally use or open them.

So I thought creating a separate file would help.

How about we link it to that shell script directly for the testing?

File: how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb
Comment (Blaizzy) [reply how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb:19]: Changed workspace. 

Let me update it. 👍

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (Blaizzy) [reply how-to-guides/reproduce-run/scripts/reproduce_run.py:22]: We have to stop the project because we are not using it anymore. I remember there is a slack thread that discussed this.

**Comment by Blaizzy** [comment]
> Some suggestions. Good work overall 🎉

Thank you very much!

File: .github/workflows/test-scripts.yml
Comment (SiddhantSadangi) [reply .github/workflows/test-scripts.yml:26]: yes, good idea.

You can append the below to this script

```bash
echo "Running run_git_checkout.sh ..."
sh run_git_checkout.sh
```

**Comment by Blaizzy** [comment]
I made changes and left some comments.

File: .github/workflows/test-scripts.yml
Comment (Blaizzy) [reply .github/workflows/test-scripts.yml:26]: Great!

Comment (Blaizzy) [reply .github/workflows/test-scripts.yml:26]: Done 👍

File: how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb
Comment (SiddhantSadangi) [inline how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb:71]:
```suggestion
    "os.environ[\"NEPTUNE_API_TOKEN\"] = \"ANONYMOUS\" # You can replace this with your own token\n"
```

File: how-to-guides/reproduce-run/scripts/reproduce_run.py
Comment (SiddhantSadangi) [reply how-to-guides/reproduce-run/scripts/reproduce_run.py:24]: Using it as a context manager automatically stops it

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
Some minor edits. but done overall.
Will wait for @normandy7's review before merging

File: how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb
Comment (normandy7) [inline how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb:373]:
```suggestion
    "Once you are done logging, stop each of the runs."
```

File: how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb
Comment (normandy7) [inline how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb:397]: Sounds a bit vague and not that useful. Maybe just leave it out? :)

**Review by normandy7 — CHANGES_REQUESTED** [review]

File: how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb
Comment (Blaizzy) [reply how-to-guides/reproduce-run/notebooks/reproduce_run.ipynb:397]: Indeed, I will remove it.

**Comment by Blaizzy** [comment]
@normandy7 @SiddhantSadangi 

All changes are implemented 👍

**Review by normandy7 — DISMISSED** [review]

**Review by SiddhantSadangi — APPROVED** [review]

**Pull request merged**
